iceoryx_utils/include/iceoryx_utils/internal/lockfree_queue/unique.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <utility>
+
+namespace iox
+{
+template <typename T>
+class unique
+{
+  public:
+    struct invalid_t
+    {
+        explicit constexpr invalid_t()
+        {
+        }
+    };
```

Comment:
I have at the moment no idea why we need this class. This is the implementation of an ```cxx::optional```. I would strongly recommend removing this class and use cxx::optional since it is fully tested and it does exactly what this class does! Additionally, why is the name called unique?